torch/_prims/__init__.py

```diff
@@ -2231,10 +2231,12 @@ def _arange_meta(
         step != 0,
         lambda: "step must be nonzero",
     )
-    utils.check(
-        math.isfinite(start) and math.isfinite(end),
-        lambda: f"unsupported range: {start} -> {end}",
-    )
+    # SymInts can't represent inf
+    if not isinstance(start, torch.SymIntNode) and not isinstance(end, torch.SymIntNode):
+        utils.check(
+            math.isfinite(start) and math.isfinite(end),
```

Comment:
FX monkeypatches math functions so that they are overrideable, and we should probably do that during make_fx tracing as well, so that a hardcode here isn't necessary